Improvement

chcmedeiros opened the pull request:
Includes a fix to prevent potential buffer over-reads in address matching logic by capping the maximum length of the address.

0xPxt commented on app/src/tx_display.c:
```diff
@@ -143,8 +143,14 @@ __Z_INLINE bool address_matches_own(char *addr) {
   }
 
+  // Cap the length to prevent buffer over-read
+  const size_t max_len =
```
Should we instead validate that `parser_tx_obj.tx_json.own_addr_len` is never greater than `MAX_BECH32_ADDR_LEN`?
I'm assuming if the length is too large, the own address is wrong or something went wrong when computing the length

chcmedeiros replied:
It is other possible way to do it, I just kept this because ledger what the cap alternative.
With the current option:
-  If own_addr_len is 200, cap to 128 and continue
- loop runs with truncated length
- address comparison fails
- returns false

with your option:
- If own_addr_len is 200, return false immediately

0xPxt replied:
I think in this case address comparison does not fail because we are only comparing up until the capped length.
Also we check lengths with the capped length.
This happens some lines below this comment :
```
  // Here actual_len has only been incremented up until max_len in the while loop
  if (strlen(addr) != actual_len) {
    return false;
  }
  if (strncmp(parser_tx_obj.tx_json.own_addr, addr, actual_len) != 0) {
    return false;
  }
```

chcmedeiros committed:
Improvement

Includes a fix to prevent potential buffer over-reads in address matching logic by capping the maximum length of the address.
